test/test_rate_limiter.js

```diff
@@ -69,18 +69,71 @@ describe('Rate limiter', function() {
 
     queue.on(
       'completed',
+      // after every job has been completed
       _.after(numJobs, function() {
         try {
-          expect(new Date().getTime() - startTime).to.be.above(
-            (numJobs - 1) * 1000
-          );
+          var timeDiff = new Date().getTime() - startTime;
+          expect(timeDiff).to.be.above((numJobs - 1) * 1000);
           done();
         } catch (e) {
-          done(e);
+          assert.fail(e);
         }
       })
     );
 
-    queue.on('failed', done);
+    queue.on('failed', function(e) {
+      assert.fail(e);
+    });
+  });
+
+  it('should put a job into the delayed queue when limit is hit', function() {
+    queue.on('failed', function(e) {
```

Comment:
Might be better to create a new queue here as you did in the test below, since this test assumes that the queue max is set to 1.